docs/src/pages/components/data-grid/scrolling/ScrollPlayground.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { Grid, Box, Button } from '@mui/material';
+import {
+  DataGridPro,
+  useGridApiRef,
+  visibleGridRowCountSelector,
+  visibleGridColumnsLengthSelector,
+  visibleGridColumnsSelector,
+  visibleSortedGridRowIdsSelector,
+  GridCellParams,
+} from '@mui/x-data-grid-pro';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+export default function ScrollPlayground() {
+  const apiRef = useGridApiRef();
+
+  const [coordinates, setCoordinates] = React.useState({
+    rowIndex: 0,
+    colIndex: 0,
+  });
+
+  const { data } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 100,
+  });
+
+  React.useEffect(() => {
+    const { rowIndex, colIndex } = coordinates;
+    apiRef.current.scrollToIndexes(coordinates);
+    const id = visibleSortedGridRowIdsSelector(apiRef.current.state)[rowIndex];
+    const column = visibleGridColumnsSelector(apiRef.current.state)[colIndex];
+    apiRef.current.setCellFocus(id, column.field);
+  }, [apiRef, coordinates]);
+
+  const handleClick = (position: string) => () => {
+    const maxRowIndex = visibleGridRowCountSelector(apiRef.current.state) - 1;
+    const maxColIndex = visibleGridColumnsLengthSelector(apiRef.current.state) - 1;
+
+    setCoordinates((coords) => {
+      if (position === 'top') {
+        return { ...coords, rowIndex: Math.max(0, coords.rowIndex - 1) };
+      }
+
+      if (position === 'bottom') {
+        return { ...coords, rowIndex: Math.min(maxRowIndex, coords.rowIndex + 1) };
+      }
+
+      if (position === 'left') {
+        return { ...coords, colIndex: Math.max(0, coords.colIndex - 1) };
+      }
+
+      return { ...coords, colIndex: Math.min(maxColIndex, coords.colIndex + 1) };
+    });
+  };
+
+  const handleCellClick = (params: GridCellParams) => {
+    const rowIndex = visibleSortedGridRowIdsSelector(apiRef.current.state).findIndex(
+      (id) => id === params.id,
+    );
+    const colIndex = visibleGridColumnsSelector(apiRef.current.state).findIndex(
+      (column) => column.field === params.field,
+    );
+    setCoordinates({ rowIndex, colIndex });
+  };
+
+  return (
+    <div style={{ width: '100%' }}>
+      <Box sx={{ width: 400, margin: '0 auto 16px' }}>
+        <Grid container justifyContent="center">
+          <Grid item>
+            <Button onClick={handleClick('top')}>top</Button>
+          </Grid>
+        </Grid>
+        <Grid container textAlign="center">
+          <Grid item xs={6}>
+            <Button onClick={handleClick('left')}>left</Button>
+          </Grid>
+          <Grid item xs={6}>
+            <Button onClick={handleClick('right')}>right</Button>
+          </Grid>
+        </Grid>
```

Comment:
A 5th button "Go back to 0, 0" could be nice, to show how it behaves with major scroll jumps.